### PR TITLE
DOMNodeList::item() returns DOMNode instead of DOMElement

### PR DESCRIPTION
**Pull request by smoench**
https://secure.php.net/manual/en/domnodelist.item.php

### CODE REVIEW

**Comment by orklah** [comment]
Seems like it can return null too


> Return Values
> 
> The node at the indexth position in the DOMNodeList, or NULL if that is not a valid index.
>

File: dom/dom_c.php [hunk 1]
```diff
@@ -1147,7 +1147,7 @@ class DOMNodeList implements Traversable, Countable {
 	 * @param int $index <p>
 	 * Index of the node into the collection.
 	 * </p>
-	 * @return DOMElement The node at the indexth position in the 
+	 * @return DOMNode The node at the indexth position in the
```
Comment (b1rdex) [inline dom/dom_c.php]:
```suggestion
	 * @return DOMNode|null The node at the indexth position in the 
```

**Comment by MaXal** [comment]
That is incorrect and should be reverted: https://3v4l.org/4Z5oj

**Comment by smoench** [comment]
@MaXal your code example is incorrect. This would be a valid one https://3v4l.org/EL06S
`DOMElement` extends from `DOMNode` -> http://php.net/DomElement
The php documents says `DOMNodeList::node()` returns `DOMNode|null` -> http://php.net/manual/en/domnodelist.item.php

**Pull request merged**